src/main/java/sirius/pasta/tagliatelle/macros/IfNotDefinedMacro.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.pasta.tagliatelle.macros;
+
+import sirius.kernel.commons.Strings;
+import sirius.kernel.di.std.Register;
+import sirius.kernel.tokenizer.Position;
+import sirius.pasta.noodle.Environment;
+import sirius.pasta.noodle.compiler.CompilationContext;
+import sirius.pasta.noodle.compiler.ir.Node;
+import sirius.pasta.noodle.macros.BasicMacro;
+import sirius.pasta.tagliatelle.rendering.GlobalRenderContext;
+import sirius.pasta.tagliatelle.rendering.LocalRenderContext;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+
+/**
+ * Checks whether the given guard is defined or not.
+ */
+@Register
+public class IfNotDefinedMacro extends BasicMacro {
+
+    @Override
+    protected Class<?> getType() {
+        return boolean.class;
+    }
+
+    @Override
+    protected void verifyArguments(CompilationContext compilationContext, Position position, List<Class<?>> args) {
+        if (args.size() != 1) {
```

Comment:
This doesn't actually check whether the argument as a string ... maybe add ` || !CompilationContext.isAssignableTo(args.getFirst(), String.class)` like in other macros?

Reply:
Should I improve it for the `isFilledMacro` aswell?
It's where I have the check from originally

Reply:
I left `isFilled` as it was to
a) don't break something accidentially
b) make sure it stays compatible with Object like the `Strings.isFilled`...

Reply:
Alternatively the error message in `isFilled` could be adapted